src/main/java/org/broadinstitute/hellbender/utils/Utils.java

```diff
@@ -515,11 +517,64 @@ public static boolean endsWith(final byte[] big, final byte[] suffix) {
      * @return the same object
      * @throws IllegalArgumentException if a {@code o == null}
      */
-    public static <T> T nonNull(T object) throws IllegalArgumentException {
+    public static <T> T nonNull(final T object) throws IllegalArgumentException {
         if (object == null) {
             throw new IllegalArgumentException("Null object is not allowed here.");
         }
         return object;
     }
 
+    /**
+     * Checks that an {@link Object} is not {@code null} and returns the same object or throws an {@link IllegalArgumentException}
+     * @param object any Object
+     * @param message the text message that would be pass to the exception thrown when {@code o == null}.
+     * @return the same object
+     * @throws IllegalArgumentException if a {@code o == null}
+     */
+    public static <T> T nonNull(final T object, final String message) throws IllegalArgumentException {
+        if (object == null) {
+            throw new IllegalArgumentException(message);
+        }
+        return object;
+    }
+
+    /**
+     * Checks whether an index is within bounds considering a collection or array of a particular size
+     * whose first position index is 0
+     * @param index the query index.
+     * @param length the collection or array size.
+     * @return same value as the input {@code index}.
+     */
+    public static int validIndex(final int index, final int length) {
+        if (index < 0) {
+            throw new IllegalArgumentException("the index cannot be negative: " + index);
+        } else if (index >= length) {
+            throw new IllegalArgumentException("the index points past the last element of the collection or array: " + index + " > " + (length -1));
+        }
+        return index;
+    }
+
+    /**
+     * Checks that a user provided file is in fact a regular (i.e. not a directory or a special device) readable file.
+     *
+     * @throws IllegalArgumentException if {@code file} is {@code null} or {@code argName} is {@code null}.
+     * @throws UserException if {@code file} is not a regular file or it cannot be read.
+     * @return the same as the input {@code file}.
+     */
+    public static File regularReadableUserFile(final File file, final String argName) {
+        nonNull(file,"unexpected null file reference");
+        nonNull(argName,"unexpected argument name");
+        if (!file.exists()) {
```

Comment:
Its was about produce a more informative message for the user... that said I guess there is not much gain in telling between reachability or permission problem... he/she can figure it out.